### PR TITLE
Adafruit Feather nrf52840 updates

### PR DESCRIPTION
**Pull request by jacobw**
The Adafruit Feather nrf52840 board was changed to Adafruit Feather recently, probably as part of the automated hwmv2 changes. The first commit gives the board the more specific name Adafruit Feather nrf52840 Express, and changes the directory to support other versions.

This second commit then adds the Adafruit Feather nrf52840 Sense, which is based on the same board but has more sensors, by making a common dts which is mostly shared.

### CODE REVIEW

File: boards/adafruit/feather_nrf52840/doc/index.rst
Comment (decsny) [inline boards/adafruit/feather_nrf52840/doc/index.rst:1]: do we need to add a code owner to this board? nobody is being added to review

Comment (jacobw) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: Assume this isn’t a question for me?

Comment (decsny) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: It's for you, or anybody who knows about this board, somebody should add themself to the code owner file or something to get pulled in to review if they have an interest in this board

Comment (jacobw) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: I’m happy to own these boards if there is no one better to do it. I’ve been running my board defs out of tree for a while and there hasn’t been much other discussion on them as far as I’ve seen.

is there something specific I need to do?

Comment (decsny) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: Just add your name to the code owner file in the top of the repo for these boards is what people usually do for stuff like this

You can do it as another commit in this PR

Comment (jacobw) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: Ok will do this evening.

Comment (jacobw) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: I added my name in another commit. Is there anything else I can do for this PR?

Comment (bjarki-andreasen) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: @decsny There is a bit of a process to be added as maintainer/collaborator, see https://docs.zephyrproject.org/latest/project/project_roles.html

File: boards/adafruit/feather_nrf52840/Kconfig.adafruit_feather_nrf52840_sense [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Adafruit Feather nRF52840 Express board configuration
+
+# Copyright (c) 2020 Tobias Svehagen
```
Comment (bjarki-andreasen) [inline boards/adafruit/feather_nrf52840/Kconfig.adafruit_feather_nrf52840_sense:3]: The copyrights seem incorrect, this is a file created by Jacub right? copied from a file by Tobias Svehagen
In this case, add your own copyright to the file as well:
```
# Copyright (c) 2020 Tobias Svehagen
# Copyright (c) 2024 Jakub W
```

Comment (jacobw) [reply boards/adafruit/feather_nrf52840/Kconfig.adafruit_feather_nrf52840_sense:3]: Yes that's correct. Ok I can amend as requested,

What's the threshold for adding a copyright? I made at least minor modifications to most of the files (i.e. in addition to renaming them). Should I amend all files I edited with a copyright?

Comment (bjarki-andreasen) [reply boards/adafruit/feather_nrf52840/Kconfig.adafruit_feather_nrf52840_sense:3]: Don't know exactly :) I just want to be on the safe side

File: boards/adafruit/feather_nrf52840/doc/index.rst
Comment (henrikbrixandersen) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: @bjarki-andreasen There's a difference between the CODEOWNERS file and the MAINTAINERS.yaml file. The first does not require a process.

Comment (bjarki-andreasen) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: @henrikbrixandersen
ok, I just read the "_This file is going to be deprecated and currently only had
entries that are not covered by the MAINTAINERS file._" section of the file, did not know we actually used it :)

Comment (decsny) [reply boards/adafruit/feather_nrf52840/doc/index.rst:1]: > @henrikbrixandersen ok, I just read the "_This file is going to be deprecated and currently only had entries that are not covered by the MAINTAINERS file._" section of the file, did not know we actually used it :)

yeah, I've seen that blurb, doesn't seem to be true, seems still the best place to put random board and driver owners and that's what I see people doing a lot

**Review by bjarki-andreasen — APPROVED** [review]
Looks good :) Nice work!

**Review by fabiobaltieri — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by nordicjm — COMMENTED** [review]
This is missing a deprecated board name entry, please add it otherwise anyone going from 3.7 or older will just have a "no such board" error

**Comment by jacobw** [comment]
> This is missing a deprecated board name entry, please add it otherwise anyone going from 3.7 or older will just have a "no such board" error

PR #76624 created to resolve this.

**Pull request merged**